app/src/main/java/net/squanchy/tweets/service/TwitterService.java

```diff
@@ -0,0 +1,129 @@
+package net.squanchy.tweets.service;
+
+import com.twitter.sdk.android.core.Callback;
+import com.twitter.sdk.android.core.Result;
+import com.twitter.sdk.android.core.TwitterException;
+import com.twitter.sdk.android.core.models.Identifiable;
+import com.twitter.sdk.android.tweetui.Timeline;
+import com.twitter.sdk.android.tweetui.TimelineResult;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import net.squanchy.tweets.view.TimelineStateHolder;
+
+public class TwitterService<T extends Identifiable> {
+
+    private static final long CAPACITY = 200L;
+    private final Timeline<T> timeline;
+    private final TimelineStateHolder timelineStateHolder;
+    private final List<T> itemList;
+
+    public TwitterService(Timeline<T> timeline) {
+        this.timeline = timeline;
+        this.timelineStateHolder = new TimelineStateHolder();
+
+        itemList = new ArrayList<>();
+    }
+
+    public void refresh(Callback<TimelineResult<T>> developerCb) {
```

Comment:
@rock3r actually we already always show the loader at the bottom, the callback is just the way Twitter does the things, but I agree on wrapping it with RxJava